src/app/components/quizList.tsx

```diff
@@ -0,0 +1,149 @@
+import Quiz, { QuizParams } from "@/app/components/quiz";
+import { useState } from "react";
+import clsx from "clsx";
+
+export interface QuizListParams {
+  className?: string;
+  quizzes: QuizParams[];
+  onFinish?: () => void;
+}
+
+export default function QuizList({
+  quizzes,
+  className,
+  onFinish,
+}: QuizListParams) {
+  const [currentQuizIndex, setCurrentQuizIndex] = useState(0);
+  const [showSummary, setShowSummary] = useState(false);
+  const quiz = quizzes[currentQuizIndex];
+
+  const [quizzesState, setQuizzesState] = useState(
```

Comment:
missing type

Reply:
From the above comment I have interpreted, that types should and can be infered. Whats the reason for explicitly typing it here?

Reply:
just primitive types are being infered (number, boolean, string). states with a  complex data structure have to typed. 

Exception `const [bla, setBla] = useState<primitive type | undefined> (undefined)` => this needs to typed

Reply:
I agree with the exception, but in this case, the type can be infered (and is infered, at least in webstorm), since the map function inferes a type itself.